### PR TITLE
Create get_story_translation_tests query

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

[Retrieve storyTranslationsTests](https://www.notion.so/uwblueprintexecs/Retrieve-storyTranslationsTests-03c53ae24da54cf6888f71b72d648472)

## Implementation description

- implemented get_story_translation_tests query
- removed unused fields from StoryTranslationNodeDTO

## Steps to test

Reset data:
1. `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data erase"`
2. `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`

Make new story translation tests:
1. [make test for user 1](http://localhost:5000/graphql?query=mutation%20CreateStoryTranslationTest%20%7B%0A%20%20createStoryTranslationTest(userId%3A%201%2C%20level%3A2%2C%20language%3A%22FRENCH%22)%7B%0A%20%20%20%20story%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20translatorId%0A%20%20%20%20%20%20storyId%0A%20%20%20%20%20%20language%0A%20%20%20%20%20%20stage%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A%0A&operationName=CreateStoryTranslationTest)
2. [make test for user 2](http://localhost:5000/graphql?query=mutation%20CreateStoryTranslationTest%20%7B%0A%20%20createStoryTranslationTest(userId%3A%202%2C%20level%3A2%2C%20language%3A%22FRENCH%22)%7B%0A%20%20%20%20story%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20translatorId%0A%20%20%20%20%20%20storyId%0A%20%20%20%20%20%20language%0A%20%20%20%20%20%20stage%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A%0A&operationName=CreateStoryTranslationTest)

Test query on Altair client
- [login](http://localhost:5000/graphql?query=%0Amutation%20LoginAdmin%20%7B%0A%20%20login%20(email%3A%22planetread%2Bangelamerkel%40uwblueprint.org%22%2C%20password%3A%22.%22)%20%7B%0A%20%20%20%20accessToken%0A%20%20%20%20refreshToken%0A%20%20%20%20id%0A%20%20%7D%0A%7D&operationName=LoginAdmin) + make query as admin, verify that both story translations are fetched
- [login](http://localhost:5000/graphql?query=%0Amutation%20LoginUser%20%7B%0A%20%20login%20(email%3A%22planetread%2Bcarlsagan%40uwblueprint.org%22%2C%20password%3A%22.%22)%20%7B%0A%20%20%20%20accessToken%0A%20%20%20%20refreshToken%0A%20%20%20%20id%0A%20%20%7D%0A%7D&operationName=LoginUser) + make query as user 1, verify that only the corresponding test for the user is fetched

Query:
```
query StoryTranslationTests {
  storyTranslationTests {
    id
    storyId
    title
    language
    stage
    youtubeLink
    description
    testGrade
    testResult
    testFeedback
    translatorName
  }
}
```

## What should reviewers focus on?

- does the query work?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/services/implementations/story_service.py [hunk 1]
```diff
@@ -229,6 +229,77 @@ def get_story_translations(
...
+    def get_story_translation_tests(
+        self,
+        user,
+        language=None,
+        level=None,
+        stage=None,
+        story_title=None,
+    ):
+        try:
+            filters = [StoryTranslation.is_test]
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:241]: neat! Im not familiar with this notation

File: backend/python/app/services/implementations/story_service.py [hunk 2]
```diff
@@ -229,6 +229,77 @@ def get_story_translations(
...
+            if language is not None:
+                filters.append(StoryTranslation.language == language)
+            if stage is not None:
+                filters.append(StoryTranslation.stage == stage)
+            if level is not None:
+                filters.append(Story.level == level)
+            if story_title is not None:
+                filters.append(Story.title.like(f"%{story_title}%"))
+            if user.role != "Admin":
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:250]: v clean solution B)

File: backend/python/app/services/interfaces/story_service.py [hunk 3]
```diff
@@ -93,6 +93,21 @@ def get_story_translations(self, language, level, stage, story_title, role_filte
...
+        """Return a list of story translation tests based on filters
+
+                :param user: UserDTO
+                :param language: language of story translation tests to filter by
+                :param level: level of story translation tests to filter by
+                :param stage: stage of story translation tests to filter by
+                :param story_title: story_title of story translation tests to filter by
+                :return: list of StoryTranslationTestResponseDTO's
+                :rtype: list of StoryTranslationTestResponseDTO's
+        's
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:107]: nit: this line looks superfluous

File: backend/python/app/tests/helpers/story_translation_helpers.py [hunk 4]
```diff
@@ -62,6 +62,33 @@ def assert_story_translation_equals_model(
     assert story_translation_response["num_content_lines"] == len(translation_contents)
 
 
+def assert_story_translation_test_equals_model(
```
Comment (gaoxk) [inline backend/python/app/tests/helpers/story_translation_helpers.py:65]: AWWWWWWWWWWWWWWWWWWWWWWWWWWWWW YEAHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH WE LOVE SUM GUD UNIT TEST HELPERS

**Review by gaoxk — APPROVED** [review]
everything worked while testing!! just a few nits n questions 😎

File: backend/python/app/services/implementations/story_service.py (hunk 1 above)
Comment (Puepis) [reply backend/python/app/services/implementations/story_service.py:241]: oh it's equivalent to `StoryTranslation.is_test == true` i believe

File: backend/python/app/tests/helpers/story_translation_helpers.py (hunk 4 above)
Comment (Puepis) [reply backend/python/app/tests/helpers/story_translation_helpers.py:65]: oh it's just bc of the extra fields like `translation_contents`, `num_translated_lines`,and `num_approved_lines` but I can try using `assert_story_translation_equals_model` also

**Pull request merged**